struqture/src/bosons/bosonic_noise_operator.rs

```diff
@@ -150,11 +150,24 @@ impl<'a> OperateOnDensityMatrix<'a> for BosonLindbladNoiseOperator {
     ///
     /// * `Ok(Some(CalculatorComplex))` - The key existed, this is the value it had before it was set with the value input.
     /// * `Ok(None)` - The key did not exist, it has been set with its corresponding value.
+    /// * `Err(StruqtureError)` - The input contained identities, which are not allowed as Lindblad operators.
+    /// # Panics
+    ///
+    /// * Internal error in BosonProduct::new
     fn set(
         &mut self,
         key: Self::Index,
         value: Self::Value,
     ) -> Result<Option<Self::Value>, StruqtureError> {
+        if key.0
+            == BosonProduct::new([], []).expect("Internal error when creating empty BosonProduct.")
```

Comment:
Because `set` returns a `Result`, the `expects` can be replaced with a `?` at no cost here (and this applies to all the `set` functions)